src/app/services/cart/cart.service.ts

```diff
@@ -0,0 +1,76 @@
+import { map, take } from 'rxjs/operators';
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject, combineLatest } from 'rxjs';
+
+import { CartProduct } from '../../models/cart-product.model';
+import { StoreService } from '../store/store.service';
+import { Product } from 'src/app/models/product.model';
+
+@Injectable()
+export class CartService {
+
+  private cart$: BehaviorSubject<CartProduct[]>;
```

Comment:
why array? the main functionallity of the cart is to add and remove so with object it is better. and its kinda obvious to use name as key and quantity as value.. read about Record